packages/components/combobox/src/combobox.ts

```diff
@@ -0,0 +1,603 @@
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+
+    if (!this.input) {
+      this.input = this.querySelector<HTMLInputElement>('input[slot="input"]') || document.createElement('input');
+      this.input.autocomplete = 'off';
+      this.input.slot = 'input';
+
+      if (!this.input.parentElement) {
+        this.append(this.input);
+      }
+    }
+
+    this.input.setAttribute('role', 'combobox');
+    this.input.setAttribute('aria-autocomplete', 'both');
+    this.input.setAttribute('aria-expanded', 'false');
+    this.input.setAttribute('aria-haspopup', 'listbox');
+
+    this.#events.listen(this.input, 'click', this.#onInputClick);
+    this.#events.listen(this.input, 'focus', this.#onFocus);
+    this.#events.listen(this.input, 'pointerdown', this.#onPointerDown);
+    this.#events.listen(this.input, 'pointerup', this.#onPointerUp);
+
+    this.#observer.observe(this, { childList: true, subtree: true });
+
+    this.setFormControlElement(this);
+
+    // Listen for i18n updates and update the validation message
+    this.#events.listen(window, LOCALE_STATUS_EVENT, this.#updateValidity);
+  }
+
+  override disconnectedCallback(): void {
+    this.#observer.disconnect();
+
+    super.disconnectedCallback();
+  }
+
+  override willUpdate(changes: PropertyValues<this>): void {
+    super.willUpdate(changes);
+
+    if (changes.has('options') || changes.has('value')) {
+      this.#updateSelected();
+    }
+  }
+
+  override updated(changes: PropertyValues<this>): void {
+    super.updated(changes);
+
+    if (changes.has('autocomplete')) {
+      this.input.setAttribute('aria-autocomplete', this.autocomplete || 'both');
+    }
+
+    if (changes.has('autocomplete') || changes.has('readonly')) {
+      this.input.readOnly = this.readonly ?? this.autocomplete === 'off';
+    }
+
+    if (changes.has('disabled')) {
+      this.input.disabled = !!this.disabled;
+    }
+
+    if (changes.has('required')) {
+      this.internals.ariaRequired = this.required ? 'true' : 'false';
+
+      this.#updateValidity();
+    }
+  }
+
+  override render(): TemplateResult {
+    return html`
+      <sl-text-field
+        @input=${this.#onInput}
+        @keydown=${this.#onKeydown}
+        @sl-blur=${this.#onTextFieldBlur}
+        @sl-change=${this.#onTextFieldChange}
+        @sl-focus=${this.#onTextFieldFocus}
+        @sl-update-state=${this.#onTextFieldUpdateState}
+        ?disabled=${this.disabled}
+        ?readonly=${this.readonly}
+        ?required=${this.required}
+        placeholder=${ifDefined(this.placeholder)}
+        size=${ifDefined(this.size)}
+      >
+        ${this.multiple && this.currentSelection.length
+          ? html`
+              <sl-tag-list
+                slot="prefix"
+                size=${ifDefined(this.size)}
+                stacked
+                .emphasis=${this.disabled ? 'bold' : 'subtle'}
+              >
+                ${this.currentSelection.map(
```

Comment:
Maybe selected items in the tag list should not be ordered alphabetically, maybe we should add at the end last added element, like in Vaadin for example? https://vaadin.com/docs/latest/components/multi-select-combo-box

Reply:
Fixed